lively.morphic/html-morph.js

```diff
@@ -271,3 +271,205 @@ export class IFrameMorph extends HTMLMorph {
+ *   - loop: true/false to play on a loop (default true)
+ *   - autoplay: true/false to play on load (default true)
+ *   - autosize:  if true, resize to video size  (default false)
+ *   - autoaspect: if true, resize to correct aspect ratio within current bounding box (default true)
+ * methods:
+ *    - startPlaying() -- start playing the video
+ *    - stopPlaying() -- stop playing the video
+ *    - rewind()  -- set the video counter to 0
+ *
+ * read-only properties:
+ *     - videoDomElement -- the DOM element for the video
+ */
+
+// new VideoMorph().openInWorld()
+export class VideoMorph extends HTMLMorph {
+  static get properties () {
+    return {
+      autoplay: {
+        defaultValue: false,
+        set (aBool) {
+          this.setVideoProperty('autoplay', !!aBool);
+        }
+      },
+      loop: {
+        defaultValue: false,
+        set (aBool) {
+          this.setVideoProperty('loop', !!aBool);
+        }
+      },
+      controls: {
+        defaultValue: false,
+        set (aBool) {
+          this.setVideoProperty('controls', !!aBool);
+        }
+      },
+      src: {
+        defaultValue: false,
+        set (srcURL) {
+          this.setVideoProperty('src', srcURL);
+        }
+      },
+      type: {
+        defaultValue: false,
+        set (videoType) {
+          this.setVideoProperty('type', videoType);
+        }
+      },
+      autosize: {
+        defaultValue: false,
+        set (aBool) {
+          this.setProperty('autosize', !!aBool);
+          this.resizeToExtentProperties();
+        }
+      },
+      autoaspect: {
+        defaultValue: true,
+        set (aBool) {
+          this.setProperty('autosize', !!aBool);
+          this.resizeToExtentProperties();
+        }
+      }
+
+    };
+  }
+
+  constructor (props) {
+    this.videoId = `video-${Math.floor(Math.random() * 1E6)}`;
+    this.constructorProps = props;
+    super(props);
+  }
+
+  setVideoProperty (propertyName, value) {
+    this.setProperty(propertyName, value);
+    this.resetHTML();
+  }
+
+  onLoad () {
+    const props = this.constructorProps;
+    const setField = (field, defaultVal) => {
+      this[field] = props && props[field] ? props[field] : defaultVal;
+    };
+    const normalizeBoolean = field => {
+      this[field] = props ? !!props[field] : false;
+    };
+
+    const bools = ['loop', 'controls', 'autoplay'];
+    bools.forEach(bool => normalizeBoolean(bool));
+    const fields = [
+      { name: 'src', defaultVal: 'https://matt.engagelively.com/assets/kaleidoscope-art-17141.mp4' },
+      { name: 'type', defaultVal: 'video/mp4' },
+      { name: 'badBrowserMessage', defaultVal: 'Your browser does not support video content' },
+      { name: 'loop', defaultVal: true },
+      { name: 'autoplay', defaultVal: true },
+      { name: 'controls', defaultVal: false },
+      { name: 'autosize', defaultVal: false },
+      { name: 'autoaspect', defaultVal: true }
```

Comment:
Can we make the defaultValues here consistent with the ones specified for the morphic properties above?

Reply:
Sure

Reply:
This is now done.